http_parser.c

```diff
@@ -435,6 +435,12 @@ enum http_host_state
   (IS_ALPHANUM(c) || (c) == '.' || (c) == '-' || (c) == '_')
 #endif
 
+/**
+ * Verify that a char is a valid visible (printable) US-ASCII
+ * character or %x80-FF
+ **/
+#define IS_HEADER_CHAR(ch)                                                     \
+  (ch == CR || ch == LF || ch == 9 || (ch > 31 && ch != 127))
```

Comment:
Does this comparison work for values >= 128? `ch` is a signed char so those values would be interpreted as -128..-1 and not valid.

Also, in https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2.2 :

```
       CHAR           = <any US-ASCII character (octets 0 - 127)>
[...]
       CTL            = <any US-ASCII control character
                        (octets 0 - 31) and DEL (127)>
[...]
       SP             = <US-ASCII SP, space (32)>
       HT             = <US-ASCII HT, horizontal-tab (9)>
[...]
HTTP/1.1 defines the sequence CR LF as the end-of-line marker for all protocol elements except the entity-body (see appendix 19.3 for tolerant applications). The end-of-line marker within an entity-body is defined by its associated media type, as described in section 3.7.

       CRLF           = CR LF

HTTP/1.1 header field values can be folded onto multiple lines if the continuation line begins with a space or horizontal tab. All linear white space, including folding, has the same semantics as SP. A recipient MAY replace any linear white space with a single SP before interpreting the field value or forwarding the message downstream.

       LWS            = [CRLF] 1*( SP | HT )

The TEXT rule is only used for descriptive field contents and values that are not intended to be interpreted by the message parser. Words of *TEXT MAY contain characters from character sets other than ISO- 8859-1 [22] only when encoded according to the rules of RFC 2047 [14].

       TEXT           = <any OCTET except CTLs,
                        but including LWS>

A CRLF is allowed in the definition of TEXT only as part of a header field continuation. It is expected that the folding LWS will be replaced with a single SP before interpretation of the TEXT value. 
```

My reading is that if a CR or LF char is present, it must be in a pair (CRLF) and then must be followed by either a SP or a HT.

But overall this does over the set of valid characters, so maybe that doesn't matter. Although I would consider adding a definition for HT instead of 9.